### ISSUE TITLE
Invalid “Duplicate thoughts” error

### ISSUE THREAD

**Issue by raineorshine** [cybersemics/em#895]
Mobile only (may be reproducible by simulating mobile by setting `isMobile` to true.)

## Steps to Reproduce

1. Create thought `a`
2. Create subthought `a/b`
3. Close keyboard
4. `cursorBack`
5. `newThought`

## Current Behavior

Alerts "Duplicate thoughts are not allowed within the same context."

## Expected Behavior

Should allow the new thought to be created since there actually are no duplicates within the same context.

**Comment by shresthabijay** [comment]
@raineorshine I was not able to reproduce this both on mobile and desktop.

**Comment by raineorshine** [comment]
Try on Safari desktop. Set `isMobile = true` in `browser.ts`. For Step 3, "Close keyboard", click on an empty area on the top of the screen to blur the selection before executing `cursorBack`.

**Comment by shresthabijay** [comment]
I have tried this on mobile Safari. I was able to reproduce it. 

https://github.com/cybersemics/em/blob/394cbff80b687031db4a0d76cc6b0618ff72f52c/src/components/Editable.tsx#L503

As you can see here `editing` is set to `false` if there is no `focusNode` or if focus is in **bullet**. On Safari mobile, when keyboard is closed, then on next render `window.getSelection()?.focusNode` is `null`. So `editing` is set to `false`.

https://github.com/cybersemics/em/blob/394cbff80b687031db4a0d76cc6b0618ff72f52c/src/components/Editable.tsx#L301

Here `editMode` is always `true` for desktop and for mobile it depends on `editing`.

https://github.com/cybersemics/em/blob/394cbff80b687031db4a0d76cc6b0618ff72f52c/src/components/Editable.tsx#L313-L335

In this side effect, `setSelectionToCursorOffset` is not called when `editMode` is `false`. Since on Safari mobile `editing` value
is set to `false` as I described above, `editMode` is also `false`. `setSelectionToCursorOffset` is not called as expected. But the logic for updating the `editingValue` is inside `onFocus`.

https://github.com/cybersemics/em/blob/394cbff80b687031db4a0d76cc6b0618ff72f52c/src/components/Editable.tsx#L526

So the `editingValue` and current selected thought is out of sync here.

https://github.com/cybersemics/em/blob/394cbff80b687031db4a0d76cc6b0618ff72f52c/src/shortcuts/newThoughtOrOutdent.tsx#L39

On trying to add new thought, the logic here in `newThoughtOrOutdent` shows alert for duplicate thoughts because the head value of the cursor is not in sync with the `editingValue`>

I think `editingValue` should be placed inside the side effect instead of `onFocus`.

**Comment by raineorshine** [comment]
Thank you for the thorough write-up! This is some of the most complicated code in the app.

> In this side effect, `setSelectionToCursorOffset` is not called when `editMode` is `false`. Since on Safari mobile `editing` value
> is set to `false` as I described above, `editMode` is also `false`. `setSelectionToCursorOffset` is not called as expected. But the logic for updating the `editingValue` is inside `onFocus`.

This part is working as expected. The Editable side effect is used to restore the selection if it gets re-rendered while the user is editing it. When `editMode` is `false` (such as when the keyboard is closed), the side effect should not restore the selection.

The selection is set in roughly three cases:
1. When the user focuses on the editable
2. When a new thought is created
3. When an Editable being edited gets re-rendered

For this issue, edit mode is set in newThought:

https://github.com/cybersemics/em/blob/dev/src/reducers/newThought.ts#L144-L149

> On trying to add new thought, the logic here in `newThoughtOrOutdent` shows alert for duplicate thoughts because the head value of the cursor is not in sync with the `editingValue`

I found that setting `editingValue` to `null` on blur fixes it: #915.

Let me know if you think that's adequate! I tested it and it works, but happy to answer any questions you might have.
